### PR TITLE
Change kuroko script formatter to rails helper and expand definition name if needed

### PR DESCRIPTION
**Pull request by eisuke**
kuroko2_script

@cookpad/dev-infra 👓 ?

### CODE REVIEW

File: app/assets/stylesheets/kuroko2/job_definitions.scss [hunk 1]
```diff
@@ -17,3 +17,9 @@
 .script-input {
   font-family: 'Inconsolata';
 }
+
+.kuroko-script {
+  span.note {
+    color: #888
```
Comment (k0kubun) [inline app/assets/stylesheets/kuroko2/job_definitions.scss]: `;` is missing

File: app/helpers/kuroko2/job_definitions_helper.rb [hunk 2]
```diff
@@ -34,5 +34,59 @@ def star_link_for(definition)
         )
       end
     end
+
+    def format_kuroko_script(script_text)
+      raw(
```
Comment (k0kubun) [inline app/helpers/kuroko2/job_definitions_helper.rb]: [nits IMO] If contents are mutli lines, I think appending `.html_safe` is simpler.

File: app/helpers/kuroko2/job_definitions_helper.rb [hunk 3]
```diff
@@ -34,5 +34,59 @@ def star_link_for(definition)
...
+
+    private
+
+    def format_wait_task(line, matched)
+      definition_names = []
+      formatted_line = line.gsub(Kuroko2::Workflow::Task::Wait::OPTION_REGEXP) { |option|
+        definition = Kuroko2::JobDefinition.find_by(id: $1.to_i)
+        if definition.present?
+          definition_names << definition.name
+          link_to(option, Kuroko2::Engine.routes.url_helpers.job_definition_path(definition.id))
```
Comment (k0kubun) [inline app/helpers/kuroko2/job_definitions_helper.rb]: This is a helper method. Are you sure that `Kuroko2::Engine.routes.url_helpers` is required here?

File: spec/helpers/job_definition_helper_spec.rb [hunk 4]
```diff
@@ -39,4 +39,21 @@
...
+
+    let(:text) do
+      <<-EOF.strip_heredoc
+        wait: #{definition1.id}/daily #{definition2.id}/daily timeout=100
+        sub_process: #{definition3.id}
+      EOF
+    end
+
+    it do
+      expect(subject).to eq("wait: <a href=\"/definitions/#{definition1.id}\">#{definition1.id}/daily</a> <a href=\"/definitions/#{definition2.id}\">#{definition2.id}/daily</a> timeout=100 <span class=\"note\"># #{definition1.name}, #{definition2.name}</span>\n<a href=\"/definitions/#{definition3.id}\">sub_process: #{definition3.id}</a> <span class=\"note\"># #{definition3.name}</span>\n")
```
Comment (k0kubun) [inline spec/helpers/job_definition_helper_spec.rb]: Why don't you use heredoc for this?

File: app/assets/stylesheets/kuroko2/job_definitions.scss (hunk 1 above)
Comment (eisuke) [inline app/assets/stylesheets/kuroko2/job_definitions.scss]: opps https://github.com/cookpad/kuroko2/pull/37/commits/148cfe0c15f82b7a37ecba90254b9af00250a922

File: app/helpers/kuroko2/job_definitions_helper.rb [hunk 5]
```diff
@@ -34,5 +34,59 @@ def star_link_for(definition)
         )
       end
     end
+
+    def format_kuroko_script(script_text)
+      raw(
+        script_text.each_line.map { |line|
+          formatted_line = ERB::Util.html_escape(line)
```
Comment (k0kubun) [inline app/helpers/kuroko2/job_definitions_helper.rb]: [IMO] If `match[:type]` exists, this escape is unnecessary and bad for performance. How about this?

```rb
matched = Kuroko2::Workflow::ScriptParser::LINE_REGEXP.match(line.chomp)
case matched && matched[:type]
when 'wait'
  format_wait_task(line, matched)
when 'sub_process'
  format_sub_process_task(line, matched)
else
  ERB::Util.html_escape(line)
end
```

Comment (eisuke) [inline app/helpers/kuroko2/job_definitions_helper.rb]: 🆒  58f9454

File: spec/helpers/job_definition_helper_spec.rb (hunk 4 above)
Comment (eisuke) [inline spec/helpers/job_definition_helper_spec.rb]: 2fb0535

File: spec/helpers/job_definition_helper_spec.rb [hunk 6]
```diff
@@ -39,4 +39,24 @@
...
+    let(:text) do
+      <<-EOF.strip_heredoc
+        wait: #{definition1.id}/daily #{definition2.id}/daily timeout=100
+        sub_process: #{definition3.id}
+      EOF
+    end
+
+    it do
+      expect(subject).to eq(<<-HTML.strip_heredoc)
+        wait: <a href=\"/definitions/#{definition1.id}\">#{definition1.id}/daily</a> <a href=\"/definitions/#{definition2.id}\">#{definition2.id}/daily</a> timeout=100 <span class=\"note\"># #{definition1.name}, #{definition2.name}</span>
```
Comment (k0kubun) [inline spec/helpers/job_definition_helper_spec.rb]: [nits] Escaping double quotes is no longer necessary.

File: app/helpers/kuroko2/job_definitions_helper.rb (hunk 3 above)
Comment (eisuke) [inline app/helpers/kuroko2/job_definitions_helper.rb]: How about this? 7c12dc1

File: app/helpers/kuroko2/job_definitions_helper.rb [hunk 7]
```diff
@@ -34,5 +34,54 @@ def star_link_for(definition)
...
+
+    def format_wait_task(line, matched)
+      definition_names = []
+      formatted_line = line.gsub(Kuroko2::Workflow::Task::Wait::OPTION_REGEXP) { |option|
+        definition = Kuroko2::JobDefinition.find_by(id: $1.to_i)
+        if definition.present?
+          definition_names << definition.name
+          link_to(option, job_definition_path(definition.id))
+        else
+          option
```
Comment (adorechic) [inline app/helpers/kuroko2/job_definitions_helper.rb]: Is `ERB::Util.html_escape` necessary for `option`? `format_sub_process_task` looks to escape similar case.

Comment (eisuke) [inline app/helpers/kuroko2/job_definitions_helper.rb]: It is not necessary for now, but it is good to escape for defensively.  dbadf14

**Comment by aamine** [comment]
すばらしすぎる

**Pull request merged**